app/scripts/lib/rpc-method-middleware/handlers/provider-authorize.js

```diff
@@ -0,0 +1,187 @@
+import { EthereumRpcError } from 'eth-rpc-errors';
+import { MESSAGE_TYPE } from '../../../../../shared/constants/app';
+import {
+  isSupportedScopeString,
+  isSupportedNotification,
+  isValidScope,
+} from './caip-25';
+
+// {
+//   "requiredScopes": {
+//     "eip155": {
+//       "scopes": ["eip155:1", "eip155:137"],
+//       "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "get_balance", "personal_sign"],
+//       "notifications": ["accountsChanged", "chainChanged"]
+//     },
+//     "eip155:10": {
+//       "methods": ["get_balance"],
+//       "notifications": ["accountsChanged", "chainChanged"]
+//     },
+//     "wallet": {
+//       "methods": ["wallet_getPermissions", "wallet_creds_store", "wallet_creds_verify", "wallet_creds_issue", "wallet_creds_present"],
+//       "notifications": []
+//     },
+//     "cosmos": {
+//       ...
+//     }
+//   },
+//   "optionalScopes":{
+//     "eip155:42161": {
+//       "methods": ["eth_sendTransaction", "eth_signTransaction", "get_balance", "personal_sign"],
+//       "notifications": ["accountsChanged", "chainChanged"]
+//   },
+//   "sessionProperties": {
+//     "expiry": "2022-12-24T17:07:31+00:00",
+//     "caip154-mandatory": "true"
+//   }
+// }
+
+const providerAuthorize = {
+  methodNames: [MESSAGE_TYPE.PROVIDER_AUTHORIZE],
+  implementation: providerAuthorizeHandler,
+  hookNames: {
+    getAccounts: true,
+  },
+};
+export default providerAuthorize;
+
+async function providerAuthorizeHandler(_req, res, _next, end, _hooks) {
+  const { requiredScopes, optionalScopes, sessionProperties, ...restParams } =
+    _req.params;
+
+  if (Object.keys(restParams).length !== 0) {
+    return end(
+      new EthereumRpcError(
+        5301,
+        'Session Properties can only be optional and global',
```

Comment:
🤔 are you saying here that any other top level keys should live in `sessionProperties`?

Reply:
I believe that's the implication of the spec 

```
Session Properties requested outside of Session Properties Object
code = 5301
message = "Session Properties can only be optional and global"
```